### PR TITLE
overlay cache: slight modification of node-is-online rules

### PR DESCRIPTION
**Pull request by jtolds**
the proposition is that we aren't always consistent at pinging nodes in a regular fashion. either the satellite was not running, or the discovery refresh was not operating at the regular schedule. we will consider a node online if it was successfully contacted in the last hour, even if it has failed most recently (and perhaps temporarily). we will also consider a node offline if our most recent interaction was successful, regardless of if the most recent interaction was recent.

 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: internal/testplanet/reconfigure.go [hunk 1]
```diff
@@ -37,3 +39,11 @@ var DisablePeerCAWhitelist = Reconfigure{
 		config.Server.UsePeerCAWhitelist = false
 	},
 }
+
+// ShortenOnlineWindow returns a `Reconfigure` that sets the NodeSelection
```
Comment (mobyvb) [inline internal/testplanet/reconfigure.go]: after the new test changes I am not sure whether we need this still

File: satellite/satellitedb/overlaycache.go [hunk 2]
```diff
@@ -99,8 +99,8 @@ func (cache *overlaycache) SelectNewStorageNodes(ctx context.Context, count int,
 		AND free_bandwidth >= ?
 		AND free_disk >= ?
 		AND (total_audit_count < ? OR total_uptime_count < ?)
-		AND last_contact_success > ?
-		AND last_contact_success > last_contact_failure`
+		AND (last_contact_success > ?
+				 OR last_contact_success > last_contact_failure)`
```
Comment (stefanbenten) [inline satellite/satellitedb/overlaycache.go]: Identation is off here

File: internal/testplanet/satellite.go [hunk 3]
```diff
@@ -104,7 +104,7 @@ func (planet *Planet) newSatellites(count int) ([]*satellite.Peer, error) {
 					UptimeCount:       0,
 					AuditCount:        0,
 					NewNodePercentage: 0,
-					OnlineWindow:      time.Hour,
+					OnlineWindow:      0,
```
Comment (stefanbenten) [inline internal/testplanet/satellite.go]: @mobyvb I am not sure if its a good idea to disable the window entirely for all tests per default.
For that reason i created the above  "Reconfigure".

File: internal/testplanet/reconfigure.go (hunk 1 above)
Comment (stefanbenten) [reply internal/testplanet/reconfigure.go]: I would prefer to use this method to have the ability to test the full "grace" period of 1hr per default.
If we set the value to 0 for all tests, we are basically adding the immediate check back, right?

File: internal/testplanet/satellite.go (hunk 3 above)
Comment (littleskunk) [reply internal/testplanet/satellite.go]: Can we just merge and release it. Update the test later (if that is needed)

**Pull request merged**